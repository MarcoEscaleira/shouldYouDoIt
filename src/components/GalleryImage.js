import React from "react";

const GalleryImage = ({ answer: { img } }) => (
  <li>
    <img src={img} alt="gallery img" className="gallery__showcase__img" />
  </li>
);

export default GalleryImage;
