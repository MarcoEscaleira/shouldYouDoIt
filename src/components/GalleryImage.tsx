import React from 'react';

interface Props {
  answer: {
    img: string
  }
}

const GalleryImage: React.FC<Props> = ({ answer: { img = '' } }) => (
  <li>
    <img src={img} alt="gallery img" className="gallery__showcase__img" />
  </li>
);

export default GalleryImage;
