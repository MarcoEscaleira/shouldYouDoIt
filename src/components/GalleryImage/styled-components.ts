import styled from 'styled-components';

export const Image = styled.img`
  width: 16rem;
  height: 16rem;
  margin: 1rem;
`;