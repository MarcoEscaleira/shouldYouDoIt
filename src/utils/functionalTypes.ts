export type NullaryFn<R> = () => R;
export type UnaryFn<A, R> = (a: A) => R;
export type BinaryFn<A, B, R> = (a: A, b: B) => R;
export type TernaryFn<A, B, C, R> = (a: A, b: B, c: C) => R;
export type QuadFn<A, B, C, D, R> = (a: A, b: B, c: C, d: D) => R;
export type QuinFn<A, B, C, D, F, R> = (a: A, b: B, c: C, d: D, f: F) => R;
export type SixFn<A, B, C, D, F, H, R> = (
  a: A,
  b: B,
  c: C,
  d: D,
  f: F,
  h: H
) => R;
