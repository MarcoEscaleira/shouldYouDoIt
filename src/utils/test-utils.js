import React from 'react';
import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import { MemoryRouter as Router } from 'react-router-dom';
import configureStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import store from '../store';

export const renderWithRedux = (ui, initialState) => {
  return {
    ...render(
      <Provider store={store(initialState)}>
        <Router>{ui}</Router>
      </Provider>
    )
  };
};

export const renderWithRouter = ui => {
  return {
    ...render(<Router>{ui}</Router>)
  };
};

const middlewares = [thunk];
export const mockStore = configureStore(middlewares);
