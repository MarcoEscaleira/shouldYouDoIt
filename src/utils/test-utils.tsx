import React from "react";
import { render } from "@testing-library/react";
import { Provider } from "react-redux";
import { MemoryRouter as Router } from "react-router-dom";
import configureStore from "redux-mock-store";
import thunk from "redux-thunk";

import store, { initialStoreState } from "../store";

export const renderWithRedux = (ui: JSX.Element, initialState?: {}) => {
  const state = {
    ...initialStoreState,
    ...initialState
  };
  return {
    ...render(
      <Provider store={store(state)}>
        <Router>{ui}</Router>
      </Provider>
    )
  };
};

export const renderWithRouter = (ui: JSX.Element) => ({
  ...render(<Router>{ui}</Router>)
});

const middlewares = [thunk];
export const mockStore = configureStore(middlewares);
