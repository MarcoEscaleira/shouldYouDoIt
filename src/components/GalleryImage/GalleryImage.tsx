import React from "react";
import { Image } from "./styled-components";

interface Props {
  answer: {
    img: string;
  };
}

const GalleryImage: React.FC<Props> = ({ answer: { img = "" } }) => (
  <li>
    <Image src={img} alt="Gallery gif" />
  </li>
);

export default GalleryImage;
