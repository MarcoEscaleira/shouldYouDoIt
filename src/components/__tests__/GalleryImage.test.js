import React from 'react';
import GalleryImage from '../GalleryImage';
import { renderWithRedux } from '../../utils/test-utils';

describe('Gallery Image test suite', () => {
  it('Should render Gallery Image correctly', () => {
    const { container } = renderWithRedux(
      <GalleryImage answer={{ img: 'myGif.gif' }} />
    );

    expect(container.firstChild).toMatchSnapshot();
  });
});
