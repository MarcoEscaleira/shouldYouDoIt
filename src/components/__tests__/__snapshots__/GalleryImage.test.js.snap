// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Gallery Image test suite Should render Gallery Image correctly 1`] = `
<li>
  <img
    alt="gallery img"
    class="gallery__showcase__img"
    src="myGif.gif"
  />
</li>
`;
